nativelib/src/main/resources/scala-native/gc/commix/Phase.c

```diff
@@ -37,23 +43,23 @@ void Phase_Init(Heap *heap, uint32_t initialBlockCount) {
     // MacOs we do not share them across processes
     // We open the semaphores and try to check the call succeeded,
     // if not, we exit the process
-    heap->gcThreads.startWorkers =
-        sem_open(startWorkersName, O_CREAT | O_EXCL, 0644, 0);
+    heap->gcThreads.startWorkers = semaphore_open(startWorkersName, 0U);
     if (heap->gcThreads.startWorkers == SEM_FAILED) {
         fprintf(stderr,
                 "Opening worker semaphore failed in commix Phase_Init\n");
         exit(errno);
     }
 
-    heap->gcThreads.startMaster =
-        sem_open(startMasterName, O_CREAT | O_EXCL, 0644, 0);
+    heap->gcThreads.startMaster = semaphore_open(startMasterName, 0U);
     if (heap->gcThreads.startMaster == SEM_FAILED) {
         fprintf(stderr,
                 "Opening master semaphore failed in commix Phase_Init\n");
         exit(errno);
     }
     // clean up when process closes
     // also prevents any other process from `sem_open`ing it
+    // Closing now semaphore on windows would cause undefined behaviour.
+#ifndef _WIN32
```

Comment:
I've found that its design difference in Posix/Windows semaphores. 
Semaphores in Posix are persistent. Calling `sem_unlink` makes the named semaphore no longer obtainable when calling `sem_open` on Posix and marks kernel object representing semaphore for deletion when it would be no longer used.  Semaphores in Commix GC are never explicitly closed, so actual semaphore would be automatically removed upon the closing process. 
In Windows, there is no equivalent for `sem_unlink`, Semaphore would be automatically removed when all handles would be closed (in our case at process exit)